pkg/server/auth/auth_test.go

```diff
@@ -385,7 +385,7 @@ func TestUserPrincipal_String(t *testing.T) {
 	// principal is logged out.
 	p := auth.NewUserPrincipal(auth.ID("testing"), auth.Groups([]string{"group1", "group2"}), auth.Token("test-token"))
 
-	if s := p.String(); s != `id="testing" groups=[group1 group2]` {
+	if s := p.String(); s != `id="testing" groups=[group1 group2] tokenLength=10` {
 		t.Fatalf("principal.String() got %s, want %s", s, `id="testing" groups=[group1 group2]`)
```

Comment:
This hasn't been updated...time to extract to a string and compare :-)

Reply:
Yes! I support we still have `.String()` as a "fallback" (?), to protect the token in case someone accidentally mis-uses and dumps it out, however its a pointer now so I'm not sure if thats a problem anymore. Will test.

We could implement `MarshalLog` to return a struct for logging though.

Reply:
We could implement `MarshalLog` the pointer on the token is to guard against accidental leakage, something printing out a Principal, perhaps bad error handling for example?

Reply:
> time to extract to a string and compare

Ah I see what you meant, good catch.